sycl/source/detail/scheduler/scheduler.cpp

```diff
@@ -266,6 +298,19 @@ Scheduler::Scheduler() {
                      /*PropList=*/{}));
 }
 
+Scheduler::~Scheduler() {
+  // By specification there are several possible sync points: buffer
+  // destruction, wait() method of a queue or event. Stream doesn't introduce
+  // any synchronization point. It is guaranteed that stream is flushed and
+  // resources are released only if one of the listed sync points was used for
+  // the kernel. Otherwise resources for stream will not be released, issue a
+  // warning in this case.
+  if (StreamBuffersPool.size() > 0)
+    printf("\nWARNING: Some commands may have not finished the execution and "
+           "not all resources were released. Please be sure that all kernels "
+           "have sycnhronization points.\n");
```

Comment:
This definitely should go to `stderr` as users will be using `stdout` for their own purposes and will be surprised to see smth new in their log.

Reply:
What is this `printf` doing in this C++ program???